[Fixes #11] JS Exception on validation in MVC

kichalla opened the pull request:
@Eilon @DamianEdwards @pranavkm 

#11

Eilon commented on bower.json:
```diff
@@ -10,7 +10,8 @@
         "*.md",
-        "*.txt"
+        "*.txt",
+	"gulpfile.js"
```
Tabs :crying_cat_face:

Eilon commented on package.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "ASP.NET",
+  "version": "0.0.0",
```
Should this be the actual version? 3.2.4 or something? @DamianEdwards ?

DamianEdwards replied:
Only if we're actually creating an NPM package, otherwise this is fine.

Eilon replied:
Oh right. Ignore then.

Eilon commented:
@DamianEdwards can you look at the gulp/bower stuff?

DamianEdwards commented:
gulp stuff looks good.

We'll need to update `bower.json` with a new version when we actually tag and release this.

kichalla commented:
good point about `bower.json`...addressed feedback. @DamianEdwards @Eilon  ...I will create a tag once I push this change...also will create a task on templates to use this...

pranavkm commented on jquery.validate.unobtrusive.js:
```diff
@@ -70,11 +70,12 @@
+        var container = error.data("unobtrusiveContainer");
 
         if (container) {
+            var replaceAttrValue = container.attr("data-valmsg-replace");
```
We use the format
```JS
var replaceAttrValue = container.attr("data-valmsg-replace"),
    replace = replaceAttrValue ? $.parseJSON(replaceAttrValue) : null;
```
throughout this file. Let's stick to that convention.

pranavkm commented:
:shipit:

kichalla commented:
13386cd1b5947d8a5d23a12b531ce3960be1eba7
